packages/components/table/src/table.js

```diff
@@ -0,0 +1,148 @@
+	'ts-table',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				rtl: { type: Boolean, reflect: true },
+				cols: { type: Array, reflect: true },
+				data: { type: Array, reflect: true }
+			};
+		}
+
+		constructor() {
+			super();
+			this.cols = [];
+			this.data = [];
+			this.rtl = false;
+		}
+
+		get tableHeader() {
+			const res = this.cols.map(
+				column =>
+					html`
+						<th class="${this.getHeaderSizeClass(column.size)} ${this.getVisibilityClass(column.visibility)}">
+							<div class="${this.getHeaderClasses(column)}">
+								${column.value}
+							</div>
+						</th>
+					`
+			);
+			return html`
+				${res}
+			`;
+		}
+
+		getHeaderClasses(column) {
+			switch (column.display) {
+				case 'right':
+					return 'display-right';
+				case 'left':
+					return 'display-left';
+				case 'center':
+					return 'display-center';
+				default:
+					return '';
+			}
+		}
+
+		getVisibilityClass(visibility) {
+			return visibility || VISIBILITY.BOTH;
+		}
+
+		getHeaderSizeClass(size) {
+			switch (size) {
+				case 'small':
+					return 'col-small';
+				case 'medium':
+					return 'col-medium';
+				case 'large':
+					return 'col-large';
+				default:
+					return '';
+			}
+		}
+
+		handleClick(row) {
+			this.dispatchCustomEvent('row-click', row.content);
+		}
+
+		get tableBody() {
+			/* eslint-disable lit/no-template-bind */
+			const res = this.data.map(row => {
+				return html`
+					<tr class="${row.isSelected ? 'selected' : ''}" @click="${() => this.handleClick(row)}">
+						${this.tableRow(row.content)}
+					</tr>
+				`;
+			});
+			/* eslint-enable */
+
+			return html`
+				${res}
+			`;
+		}
+
+		tableRow(content) {
+			return this.cols.map(
+				column =>
+					html`
+						<td class="${this.getVisibilityClass(column.visibility)}">
+							${this.getItem(content[column.property], column)}
+						</td>
+					`
+			);
+		}
+
+		getItem(item, column) {
+			switch (column.type) {
+				case 'document':
+					return html`
+						<ts-document-card
+							name="${item.name}"
+							description="${item.description}"
+							mobile-description="${item.mobileDescription}"
+						></ts-document-card>
+					`;
+				case 'status':
+					return html`
+						<div class="status-cell">
+							<ts-status status="${item.status}" text="${item.text}"></ts-status>
+						</div>
+					`;
+				default:
+					const hasCustomRender = column.renderFunction && typeof column.renderFunction === 'function';
+					return html`
+						<div class="cell ${this.getHeaderClasses(column)}">
+							${hasCustomRender ? column.renderFunction(item) : item}
+						</div>
+					`;
+			}
+		}
+
+		render() {
+			return html`
+				<table>
+					<thead>
+						<tr>
+							${this.tableHeader}
+						</tr>
+					</thead>
+					<tbody>
+						${this.tableBody}
+					</tbody>
+				</table>
+				<div class="pagination">PAGINATION PLACEHOLDER</div>
```

Comment:
What is the reason for this placeholder? Can it be removed now?

Reply:
my current task doesn't require the pagination but the next task it will. 
so, if I remove it now I will have to add it again in the next iteration.
should I remove it now?

Reply:
I added `display: none;` for now on pagination.